parquet/src/main/java/org/apache/iceberg/parquet/ParquetValueWriters.java

```diff
@@ -313,6 +327,44 @@ public void write(int repetitionLevel, ByteBuffer buffer) {
     }
   }
 
+  private static class FixedBufferWriter extends PrimitiveWriter<ByteBuffer> {
+    private final int length;
+
+    private FixedBufferWriter(ColumnDescriptor desc) {
+      super(desc);
+      this.length = desc.getPrimitiveType().getTypeLength();
+    }
+
+    @Override
+    public void write(int repetitionLevel, ByteBuffer buffer) {
+      Preconditions.checkArgument(
+          buffer.remaining() == length,
+          "Cannot write byte buffer of length %s as fixed[%s]",
+          buffer.remaining(),
+          length);
+      column.writeBinary(repetitionLevel, Binary.fromReusedByteBuffer(buffer));
+    }
+  }
+
+  private static class FixedWriter extends PrimitiveWriter<byte[]> {
```

Comment:
I just moved this class from `BaseParquetWriter`, it was missing validation for length. I added it.